homie-influx/src/config.rs

```diff
@@ -1,105 +1,158 @@
+use eyre::Report;
 use influx_db_client::reqwest::Url;
 use influx_db_client::Client;
 use rumqttc::MqttOptions;
 use rustls::ClientConfig;
+use serde::{de, Deserialize, Deserializer};
 use stable_eyre::eyre;
 use stable_eyre::eyre::WrapErr;
-use std::fs::File;
-use std::io::{BufRead, BufReader};
+use std::fmt::Display;
+use std::fs::read_to_string;
+use std::str::FromStr;
 use std::sync::Arc;
 
 const DEFAULT_MQTT_CLIENT_PREFIX: &str = "homie-influx";
 const DEFAULT_MQTT_HOST: &str = "test.mosquitto.org";
 const DEFAULT_MQTT_PORT: u16 = 1883;
 const DEFAULT_INFLUXDB_URL: &str = "http://localhost:8086";
-const DEFAULT_MAPPINGS_FILENAME: &str = "mappings.conf";
+const CONFIG_FILENAME: &str = "homie_influx.toml";
+const DEFAULT_MAPPINGS_FILENAME: &str = "mappings.toml";
+
+#[derive(Clone, Debug, Default, Deserialize)]
+#[serde(default)]
+pub struct Config {
+    pub mqtt: MqttConfig,
+    pub homie: HomieConfig,
```

Comment:
If the `[homie]` section is missing, it defaults to `mappings_filename="mappings.toml"`?

Reply:
Yes.